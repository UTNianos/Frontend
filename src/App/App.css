@import '~antd/dist/antd.css';

.App {
  text-align: center;
  font-size: 20px;
}

.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-intro {
  font-size: large;
}

.ant-layout {
  background: white;
}

.Visible
{
  visibility: visible;
}

.Invisible {
  visibility: hidden;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
