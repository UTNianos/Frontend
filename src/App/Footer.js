import React from 'react';
import cssModules from 'react-css-modules';
import styles from './Footer.scss';

const Footer = () => (
  <div styleName="footer">
    <hr />
  </div>
);

export default cssModules(Footer, styles, { allowMultiple: true });