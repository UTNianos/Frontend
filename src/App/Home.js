import React from 'react';

const Home = () => (
  <div>
    <h1>Bienvenido a la nueva plataforma de UTNianos</h1>
  </div>
);

export default Home;
