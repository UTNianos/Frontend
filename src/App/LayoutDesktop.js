import React from 'react';
import Navbar from '../Navbar/Container';
import { Layout, Affix } from 'antd';
import './App.css';

const {Content, Header } = Layout;

const headerStyle = {
  width: '100%',
  height: '100%',
  background: '#fff',
  borderBottom: '1px solid gainsboro',
  padding: '0px'
};

const contentStyle = {
  padding: '50px',
};

const LayoutDesktop = ({children}) => {

  return (
  <Layout>

	  <Header style={headerStyle}>
      <Affix>
	     <Navbar mobile={false} />
      </Affix>
	  </Header>

	  <Content style={contentStyle}>
	    {children}
	  </Content>

  </Layout>
  );

}

export default LayoutDesktop;
