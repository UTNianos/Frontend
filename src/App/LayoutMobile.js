import React from 'react';
import Navbar from '../Navbar/Container';
import { Layout, Affix } from 'antd';
import './App.css';

const {Content, Header } = Layout;

const headerStyle = {
  width: '100%',
  height: '100%',
  background: '#fff',
  borderBottom: '1px solid gainsboro',
  padding: '0px'
};

const contentStyle = {
  paddingTop: '50px',
  paddingLeft: '0px',
  paddingRight: '0px'
};

const LayoutMobile = ({children}) => {

  return(
  <Layout>

    <Header style={headerStyle}>
      <Affix>
        <Navbar mobile={true} />
      </Affix>
    </Header>

    <Content style={contentStyle} className="mobile">
      {children}
    </Content>

 </Layout>
 );

}


export default LayoutMobile;
