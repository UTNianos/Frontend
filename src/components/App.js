import React from 'react';
import Footer from './Footer';
import NavbarUTNianos from '../containers/NavbarUTNianos';

const App = ({children}) => (    	  
     <div>	
		  <NavbarUTNianos />
		  {children}
		  <Footer />
	 </div>  
);

export default App;