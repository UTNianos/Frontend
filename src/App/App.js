/* eslint-disable react/no-children-prop */
import React from 'react';
import MediaQuery from 'react-responsive';
import LayoutDesktop from './LayoutDesktop';
import LayoutMobile from './LayoutMobile';

const App = ({ children }) => (
  <div>
    <MediaQuery minDeviceWidth={1224}>
      <LayoutDesktop children={children} />
    </MediaQuery>
    <MediaQuery maxDeviceWidth={1224}>
      <LayoutMobile children={children} />
    </MediaQuery>
  </div>
)

export default App;
