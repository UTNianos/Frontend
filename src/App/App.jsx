import React from 'react';
import LayoutDesktop from './LayoutDesktop';
import LayoutMobile from './LayoutMobile';
import MediaQuery from 'react-responsive';

const App = ({children}) => {

  return (
  <div>
   <MediaQuery minDeviceWidth={1224} >
      <LayoutDesktop children={children} />
   </MediaQuery>
	 <MediaQuery maxDeviceWidth={1224}>
      <LayoutMobile children={children} />
   </MediaQuery>
  </div>
  );

}

export default App;
