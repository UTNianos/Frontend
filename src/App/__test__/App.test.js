import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store' //ES6 modules
import App from '../App';
import Home from '../Home';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

const middlewares = []
const mockStore = configureStore(middlewares)

jest.mock('react-dom');
jest.mock('antd');

describe('<App />', () => {

  it('Render main app.', () => {
   const initialState = {};
   const store = mockStore(initialState);
   const tree = renderer.create(
     <Provider store={store} >
        <App children={null} />
     </Provider>
   ).toJSON();

   expect(tree).toMatchSnapshot();
  })


  it('<Home />', () => {
    const home = renderer.create(
    <MemoryRouter>
      <Home />
    </ MemoryRouter>
    ).toJSON();
    expect(home).toMatchSnapshot();
  })

});
