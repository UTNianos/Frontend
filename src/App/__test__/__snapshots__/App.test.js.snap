// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> <Home /> 1`] = `
<div
  className="main-body"
>
  <a
    href="/seguidor"
    onClick={[Function]}
  >
    <p>
      Seguidor de carrera
    </p>
  </a>
</div>
`;

exports[`<App /> Render main app. 1`] = `<div />`;
